samples/client/petstore/tiny/cpp/lib/Models/ApiResponse.h

```diff
@@ -0,0 +1,80 @@
+
+/*
+ * ApiResponse.h
+ *
+ * Describes the result of uploading an image resource
+ */
+
+#ifndef _ApiResponse_H_
+#define _ApiResponse_H_
+
+
+#include <string>
+#include "bourne/json.hpp"
+#include "Helpers.h"
+
+namespace Tiny {
+
+
+/*! \brief Describes the result of uploading an image resource
+ *
+ *  \ingroup Models
+ *
+ */
+
+class ApiResponse{
+public:
+
+    /*! \brief Constructor.
+	 */
+    ApiResponse();
+    ApiResponse(std::string jsonString);
+
+
+    /*! \brief Destructor.
+	 */
+    virtual ~ApiResponse();
+
+
+    /*! \brief Retrieve a bourne JSON representation of this class.
+	 */
+    bourne::json toJson();
+
+
+    /*! \brief Fills in members of this class from bourne JSON object representing it.
+	 */
+    void fromJson(std::string jsonObj);
+
+	/*! \brief Get 
+	 */
+	int getCode();
+
+	/*! \brief Set 
+	 */
+	void setCode(int  code);
+	/*! \brief Get 
+	 */
+	std::string getType();
+
+	/*! \brief Set 
+	 */
+	void setType(std::string  type);
+	/*! \brief Get 
+	 */
+	std::string getMessage();
+
+	/*! \brief Set 
+	 */
+	void setMessage(std::string  message);
+
+
+    private:
+	int code;
```

Comment:
We could use

```
int code{};
...
std::string type{};
```

This will zero-initialize primitive variables - with it we could skip having to initialize it explicitly in the constructor.

Reply:
If this is primitive types, does it call the defualt constructor for classes like std::string of std::list?

Reply:
Yeah, it will work for stuff that has constructors too. 
I think there is only one case where `{}` isn't enough - if you want to initialize a field using a parameter passed to constructor, eg.

```
class Something
{
public:
   Something(int some_important_parameter) : important_parameter(some_important_parameter) {}

private:
   int important_parameter; // you can't do something like '{some_important_parameter}'
};
```

Reply:
OKay cool, thanks 🍰